common_reference_generator/src/test/java/uk/gov/dhsc/htbhf/reference/ReferenceGeneratorTest.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.dhsc.htbhf.reference;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+class ReferenceGeneratorTest {
+
+    @ParameterizedTest
+    @MethodSource("referenceLengths")
+    public void testGenerateReference(int size) {
+        String reference = ReferenceGenerator.generateReference(size);
+        assertThat(reference).isNotNull();
+        assertThat(reference.length()).isEqualTo(size);
+        assertThat(reference).inHexadecimal();
+    }
+
+    private static Stream<Integer> referenceLengths() {
+        return Stream.of(
+                0,
```

Comment:
Should 0 length of reference be allowed? should probably return an error in that case